test/specs/rules/require-yoda-conditions.js

```diff
@@ -3,84 +3,37 @@ var assert = require('assert');
         checker.registerDefaultRules();
-        checker.configure({ requireYodaConditions: true });
-    });
-
-    it('should report yoda condition for strict equality', function() {
-        assert(
-            checker.checkString(
-                'if (x === 1) {\n' +
-                    'x++;\n' +
-                '}'
-            ).getErrorCount() === 1
-        );
-    });
-
-    it('should report yoda condition for unequality', function() {
-        assert(
-            checker.checkString(
-                'if (x != 1) {\n' +
-                    'x++;\n' +
-                '}'
-            ).getErrorCount() === 1
-        );
     });
 
-    it('should report yoda condition for strict unequality', function() {
-        assert(
-            checker.checkString(
-                'if (x !== 1) {\n' +
-                    'x++;\n' +
-                '}'
-            ).getErrorCount() === 1
-        );
-    });
-
-    it('should report yoda condition for gt', function() {
-        assert(
-            checker.checkString(
-                'if (x > 1) {\n' +
-                    'x++;\n' +
-                '}'
-            ).getErrorCount() === 1
-        );
-    });
+    operators.forEach(function(operator) {
+        var yodaCondition = 'var test; 1' + operator + 'test';
+        var notYodaCondition = 'var test; test' + operator + '1';
 
-    it('should report yoda condition for gte', function() {
-        assert(
-            checker.checkString(
-                'if (x >= 1) {\n' +
-                    'x++;\n' +
-                '}'
-            ).getErrorCount() === 1
-        );
-    });
+        [[operator], true].forEach(function(value) {
 
-    it('should report yoda condition for lt', function() {
-        assert(
-            checker.checkString(
-                'if (x < 1) {\n' +
-                    'x++;\n' +
-                '}'
-            ).getErrorCount() === 1
-        );
-    });
+            it('should not report yoda condition for ' + yodaCondition + ' with ' + value + ' option',
+                function() {
+                    checker.configure({ requireYodaConditions: value });
+                    assert(checker.checkString(yodaCondition).isEmpty());
+                }
+            );
 
-    it('should report yoda condition for lte', function() {
-        assert(
-            checker.checkString(
-                'if (x <= 1) {\n' +
-                    'x++;\n' +
-                '}'
-            ).getErrorCount() === 1
-        );
+            it('should report normal condition for ' + notYodaCondition + ' with ' + value + ' option',
+                function() {
+                    checker.configure({ requireYodaConditions: value });
+                    assert(checker.checkString(notYodaCondition).getErrorCount() === 1);
+                }
+            );
+        });
     });
 
     it('should report yoda condition for numeric', function() {
+        checker.configure({ requireYodaConditions: true });
```

Comment:
Usually, we making several `describe` blocks with different checker configurations. See https://github.com/jscs-dev/node-jscs/blob/master/test/specs/rules/disallow-space-after-object-keys.js